test/config/config.test.js

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict';
+
+const test = require('tape');
+
+const { normalize } = require('../../lib/config/config');
+const { CONFIG_SCHEMA, getDefaultOptions } = require('../../lib/config/schema');
+
+const { runTestFixtures } = require('../_utils');
+
+const defaultOptions = getDefaultOptions();
+
+// ---- support function
+/**
+ * Changes the keys from ELASTIC_APM_* the the proper confgi name
+ * @param {Record<string, string>} envVars
+ * @returns {Record<string, string>}
+ */
+function envToOptions(envVars) {
+  return CONFIG_SCHEMA.reduce((acc, def) => {
+    if (def.envVar && def.envVar in envVars) {
+      acc[def.name] = envVars[def.envVar];
+    }
+    return acc;
+  }, {});
+}
+
+/**
+ * Returns only the lines with a given prefix `use-agent log:``
+ * @param {string} stdout
+ * @returns {string[]}
+ */
+function getUseAgentLogs(stdout) {
+  const prefix = 'use-agent log:';
+  return stdout
+    .split('\n')
+    .filter((l) => l.startsWith(prefix))
+    .map((l) => l.replace(prefix, ''));
+}
+
+// ---- tests
+
+/** @type {import('../_utils').TestFixture[]} */
+const testFixtures = [
+  {
+    name: 'use agent - shoud have defaults',
+    script: 'fixtures/use-agent.js',
+    cwd: __dirname,
+    noConvenienceConfig: true, // we want full control of the env
+    timeout: 20000, // sanity guard on the test hanging
+    maxBuffer: 10 * 1024 * 1024, // This is big, but I don't ever want this to be a failure reason.
```

Comment:
nit: Similarly, I wonder if we should bump the default here. The default maxBuffer (in Node.js's spawn/exec) is 1MiB. I don't know if we get close to that typically. I *do* recall hitting it once in one test when I was turning on lots of debug/trace logging in the fixture, which is why I had that  `This is big, but I don't ever want this to be a failure reason.` case.   (This can be done in a separate PR.)